### PR TITLE
4967 fix deepedit AddRandomGuidanceDeepEditd transform

### PR DESCRIPTION
**Pull request by wojiazaiyugang**
Fixes https://github.com/Project-MONAI/MONAI/issues/4967

### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by diazandr3s — APPROVED** [review]
Looks good to me. Thanks!

Edited: Sorry, I didn't realise you've added this line `d[self.guidance_key] = self.guidance`. 
Have you tried to train DeepEdit on MONAI Label? These transforms are interconnected and followed a way of passing the guidance.

**Comment by wyli** [comment]
please sign off the commit messages (https://github.com/Project-MONAI/MONAI/blob/dev/CONTRIBUTING.md#signing-your-work)

File: monai/apps/deepedit/transforms.py [hunk 1]
```diff
@@ -595,20 +595,20 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
...
+                                self.guidance[key_label] = []
                         logger.info(f"Number of simulated clicks: {counter}")
                         break
 
                 # Breaking once all labels are covered
                 if len(keep_guidance) == len(d["label_names"].keys()):
                     logger.info(f"Number of simulated clicks: {counter}")
                     break
-
+        d[self.guidance_key] = self.guidance
```
Comment (diazandr3s) [inline monai/apps/deepedit/transforms.py:611]: I'd suggest we first perform training and inference on MONAI Label. This might break the training

**Review by diazandr3s — CHANGES_REQUESTED** [review]

File: monai/apps/deepedit/transforms.py (hunk 1 above)
Comment (wojiazaiyugang) [reply monai/apps/deepedit/transforms.py:611]: image

this is from training

Comment (wojiazaiyugang) [reply monai/apps/deepedit/transforms.py:611]: image

this is from monai label inference, when inference, AddRandomGuidanceDeepEditd is not used.
AddRandomGuidanceDeepEditd only used in training to add some random simulation guidance

Comment (wojiazaiyugang) [reply monai/apps/deepedit/transforms.py]: image

this is from monai label deepedit training, same as train in monai

**Comment by wojiazaiyugang** [comment]
I noticed that there is one unit test (tests/test_deepedit_transforms.py/TestAddRandomGuidanceCustomd) that fails, but I don't know how to fix it. My current thinking is that AddRandomGuidanceDeepEditd should add some random points, but these are randomly generated so I can't unit test. Please point out if my understanding is wrong, thanks

**Comment by diazandr3s** [comment]
> I noticed that there is one unit test (tests/test_deepedit_transforms.py/TestAddRandomGuidanceCustomd) that fails, but I don't know how to fix it. My current thinking is that AddRandomGuidanceDeepEditd should add some random points, but these are randomly generated so I can't unit test. Please point out if my understanding is wrong, thanks

Sorry, I'm a bit lost here. Why are we modifying the DeepEdit transforms? What's your use case?

**Comment by wojiazaiyugang** [comment]
> > I noticed that there is one unit test (tests/test_deepedit_transforms.py/TestAddRandomGuidanceCustomd) that fails, but I don't know how to fix it. My current thinking is that AddRandomGuidanceDeepEditd should add some random points, but these are randomly generated so I can't unit test. Please point out if my understanding is wrong, thanks
> 
> Sorry, I'm a bit lost here. Why are we modifying the DeepEdit transforms? What's your use case?

Please refer to this issue
https://github.com/Project-MONAI/MONAI/issues/4967

To sum up, I tracked the code when I trained DeepEdit and found that the AddRandomGuidanceDeepEditd transform did not add some random guidance as expected. I think may forgot to update `d` at the end.

**Comment by diazandr3s** [comment]
> > > I noticed that there is one unit test (tests/test_deepedit_transforms.py/TestAddRandomGuidanceCustomd) that fails, but I don't know how to fix it. My current thinking is that AddRandomGuidanceDeepEditd should add some random points, but these are randomly generated so I can't unit test. Please point out if my understanding is wrong, thanks
> > 
> > 
> > Sorry, I'm a bit lost here. Why are we modifying the DeepEdit transforms? What's your use case?
> 
> Please refer to this issue #4967
> 
> To sum up, I tracked the code when I trained DeepEdit and found that the AddRandomGuidanceDeepEditd transform did not add some random guidance as expected. I think may forgot to update `d` at the end.

Thanks for this, @wojiazaiyugang 
I've completely missed this bit. This will significantly help DeepEdit performance

**Comment by wyli** [comment]
could you please have a look @SachidanandAlle? it seems there's still an issue with the windows tests (https://github.com/Project-MONAI/MONAI/runs/8096979079?check_suite_focus=true) @diazandr3s

**Comment by diazandr3s** [comment]
> could you please have a look @SachidanandAlle? it seems there's still an issue with the windows tests (https://github.com/Project-MONAI/MONAI/runs/8096979079?check_suite_focus=true) @diazandr3s

Just fix the issue. I think this time it will pass :)

**Pull request merged**